### PR TITLE
Centralize reused spacings in theme.

### PR DESCRIPTION
**Pull request by linuspahl**
## Motivation and Context

This PR is creating a new section for the styled components theme called
spacings. By centralizing the spacings in the theme we can easily reuse them whenever we need to. It will
also make it easier to define margins and paddings relative to our commonly used spacings.

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/theme/margins.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import PropTypes from 'prop-types';
+
+export type Margins = {
```
Comment (linuspahl) [inline graylog2-web-interface/src/theme/margins.ts:19]: Maybe you have a better name in mind than `margins`? I thought about others like `spacings` or `gabs`, but imo `margins` is more descriptive.

**Review by kyleknighted — CHANGES_REQUESTED** [review]
I like this concept! I think `Spacing` or `Gap` is probably a better name so that way we’re not setting padding from something called margins

Also, would like to assess some generated responsive spacing so that way we can have either something like “spacing-sm, spacing-med” etc
Or we could go Tailwind-esk with `spacing-N` and generate a number of sizes from minuscule to huge that can be used

I would also like to avoid defining `px` within the css and instead have the theme handle the scale.

File: graylog2-web-interface/src/theme/margins.ts (hunk 1 above)
Comment (linuspahl) [reply graylog2-web-interface/src/theme/margins.ts]: I renamed the section to `spacings`.

**Comment by linuspahl** [comment]
Closing in favour of https://github.com/Graylog2/graylog2-server/pull/10719

**Pull request closed without merging**